src/generator.cs

```diff
@@ -5854,7 +5988,7 @@ public void Generate (Type type)
 						print ("return Dlfcn.GetSizeF (Libraries.{2}.Handle, \"{1}\");", field_pi.Name, fieldAttr.SymbolName, library_name);
 					} else if (field_pi.PropertyType == typeof (long)){
 						print ("return Dlfcn.GetInt64 (Libraries.{2}.Handle, \"{1}\");", field_pi.Name, fieldAttr.SymbolName, library_name);
-#if !WATCH
+#if HAVE_COREMEDIA && HAVE_AVFOUNDATION
```

Comment:
Yes, I agree the code inside the ifdef should be refactored, but not in this PR 😄 

Right now every platform has AVFoundation, and every platform except watchOS has CoreMedia.